node.go

```diff
@@ -247,40 +243,20 @@ func (n *Node) Start(peers []string) error {
 		return ErrStopped
 	}
 
-	_, err := n.ml.Join(peers)
-	if err != nil {
-		return fmt.Errorf("failed to join memberlist: %w", err)
-	}
-
-	// Force ourselves back into the pending state. This MUST be done after the
-	// join: join does a state sync and updates our lamport clock.
-	// NOTE: We delay taking stateMut here, because under certain conditions,
-	// it may be contested with another call when merging remote state inside
-	// of Join, leading to a deadlock.
-	n.stateMut.Lock()
-	defer n.stateMut.Unlock()
-	if err := n.changeState(peer.StateViewer, nil); err != nil {
-		return err
-	}
-
-	// Join won't return until we've done a push/pull; if there was a name
-	// collision during the join, there will be an enqueued element in the
-	// conflict queue.
-	conflict, ok := n.conflictQueue.TryDequeue()
-	if ok {
-		_ = n.ml.Shutdown()
-
-		conflict := conflict.(*memberlist.Node)
-		return fmt.Errorf("failed to join memberlist: name conflict with %s", conflict.Address())
-	}
-
```

Comment:
AFAICT the name uniqueness is not something only we enforce here, but also memberlist
https://github.com/hashicorp/memberlist/blob/e5eb6d78d6f23ad6ee996966b5c6c3f116f1b510/state.go#L1059-L1071

The conflicting name will only be allowed if the old node is dead and its reclaim time has passed, and I'm not sure if the state will be correctly merged inside the Join call.

Reply:
Makes sense to me. Do you think there's anything else for us to change here, in that case?

Reply:
Just trying to wrap my head around how that will work in practice. I'm not 100% sure that the case we discussed (that both nodes enter the ring but calculate the same work for themselves) is going to ever be true.

In that case, I think it's the first node that enters the one that participates and the other is in a split-brain state.
Or in the case that the two conflicting nodes end up communicating with a different peer subset, I think the state won't  ever be able to get merged, but each subset of the nodes only going to see one of the two conflicting names.

Reply:
Yeah, that sounds right: the cluster will never fully converge because there's nodes with overlapping names, and some of them may assign different work to different duplicates.